### ISSUE TITLE
[OJS] Crossref XML export outputting empty title metadata due to NULL locale

### ISSUE THREAD

**Issue by davidjb** [pkp/pkp-lib#6897]
**Describe the bug**
Exporting or depositing certain articles to Crossref from OJS causes the following error (akin to https://github.com/pkp/pkp-lib/issues/6021):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<doi_batch_diagnostic status="completed" sp="ds4.crossref.org">
   <submission_id>1494104596</submission_id>
   <batch_id>_1628173364</batch_id>
   <record_diagnostic status="Failure" msg_id="56">
      <doi>none</doi>
      <msg>Deposit contains title error(s): Title name value is missing from the title element</msg>
   </record_diagnostic>
   <batch_data>
      <record_count>1</record_count>
      <success_count>0</success_count>
      <warning_count>0</warning_count>
      <failure_count>1</failure_count>
   </batch_data>
</doi_batch_diagnostic>
```

caused by the XML output looking like this:

```xml
[...]
  <journal_article publication_type="full_text" metadata_distribution_opts="any">
    <titles>
      <title/>
    </titles>
[...]
```

Only specific article XML exports have this issue, some on the same journals have correct titles. The publications/submissions themselves all have titles and the Crossref deposits have previously worked (though it has been an on earlier version of OJS, pre-3.3.0-4).

**To Reproduce**
1. Go to Crossref Plugin
2. Click on Articles
3. Select an article and export XML
4. See that `<title/>` may be empty

**What application are you using?**
 OJS version 3.3.0-4

**Additional information**

`$locale` is NULL at this point of outputting the title in the filter (https://github.com/pkp/ojs/blob/main/plugins/importexport/crossref/filter/ArticleCrossrefXmlFilter.inc.php#L88) for the articles that have the issue, but the data held in the `$publication` object is in associative arrays with an `en_US` key: `["title"]=> array(1) { ["en_US"]=> string(6) "Some article title" }`.

Using `$publication->getLocalizedData()` gets the available data correctly, so it's an issue of either the publication missing its locale or the code needing to fall back, using `getLocalizedData`. For note, the working articles have their locale set as `en_US` rather than NULL, so that confirms the issue.  I imagine this can be fixed in the database but I just don't currently know where `$locale = $publication->getData('locale');` gets its data from.

```ini
[i18n]
locale = en_US 
```
is configured in my `config.inc.php` file.

**Comment by NateWr** [comment]
In the database, the `submissions` table should have a `locale` column with a value for every submission. If there are any with an empty `locale` that will cause a problem.

**Comment by davidjb** [comment]
Great, thanks @NateWr.  That's what I was missing -- was looking for the locale set over in either the `publications` or `publication_settings` tables because the filter appeared to be looking up the locale on the publication.  

All of the submissions in my database that were created in or before OJS 3.2.1.2 had the `locale` set as NULL, so doing the following resolved the issue:

```sql
UPDATE submissions SET locale = 'en_US' WHERE locale is NULL;
```

Anything published from 3.3.0.3 and above had a locale set and thus no issues with missing titles.

Would it be worth adding a migration step to handle this issue for submissions that have been created before v3.3?

**Comment by NateWr** [comment]
> Would it be worth adding a migration step to handle this issue for submissions that have been created before v3.3?

I suspect that this is a fairly isolated problem, because if it was more widespread I think we would have encountered it before. If we hear of more instances running into this we'll definitely implement a check during migration.
